ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Mock/UTxO.hs

```diff
@@ -50,22 +52,31 @@ import           Ouroboros.Consensus.Ledger.Mock.Address
   Basic definitions
 -------------------------------------------------------------------------------}
 
-data Tx = UnsafeTx (Set TxIn) [TxOut]
+data Expiry
+  = DoNotExpire
+  | ExpireAtOnsetOf !SlotNo
+  deriving stock    (Show, Eq, Ord, Generic)
+  deriving anyclass (Serialise, NFData, NoUnexpectedThunks)
```

Comment:
Why is `NFData` (and that orphan) needed?

Reply:
`Tx` derives it, and `Expiry` occurs in `Tx`. I didn't question why `Tx` derives it.